### PR TITLE
Add support for IBM Cloud Service Credentials secret type

### PR DESCRIPTION
**Pull request by nitzann**
## Problem Statement

IBM Cloud's Secret Manager service added a new secret type called [Service Credentials](https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-service-credentials&interface=ui).

## Related Issue

[Feature request](https://github.com/external-secrets/external-secrets/issues/2947)

## Proposed Changes

We would like to add the support for this secret kind in ESO, for the use of IBM Cloud users who need to manage the life cycle of service credentials within Secrets Manager, and want to use ESO to pull them into Kubernetes secrets

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: go.mod
Comment (Shanti-G) [inline go.mod:1]: This can be removed I suppose.

File: go.mod [hunk 1]
```diff
@@ -234,3 +234,5 @@ require (
 	k8s.io/klog/v2 v2.110.1 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 )
+
+//replace github.com/IBM/secrets-manager-go-sdk/v2 => /Users/nitzann/projects/SM/secrets-manager-go-sdk
```
Comment (Shanti-G) [inline go.mod:238]: This can be reviewed I suppose

File: pkg/provider/ibm/provider_test.go [hunk 2]
```diff
@@ -1164,7 +1216,9 @@ func TestGetSecretMap(t *testing.T) {
 				t.Errorf("unexpected error: %s, expected: '%s'", err.Error(), v.expectError)
 			}
 			if err == nil && !reflect.DeepEqual(out, v.expectedData) {
-				t.Errorf("unexpected secret data: expected:\n%+v\ngot:\n%+v", v.expectedData, out)
+				exp := bytes.NewBuffer(v.expectedData["credentials"]).String()
+				actual := bytes.NewBuffer(out["credentials"]).String()
+				t.Errorf("unexpected secret data: expected:\n%+v\nexp:%s\ngot:\n%+v\nactual:%s", v.expectedData, exp, out, actual)
```
Comment (Shanti-G) [inline pkg/provider/ibm/provider_test.go:1221]: Is it required to specifically extract credentials key?
If not, can this change be reverted?

**Review by Shanti-G — APPROVED** [review]

**Comment by Skarlso** [comment]
@nitzann Hi! 👋 

Would you be able to provide some outputs and screenshots for a manual run using this new feature please? I don't have an IBM cloud service account. :)

File: pkg/provider/ibm/provider.go [hunk 3]
```diff
@@ -294,6 +299,26 @@ func getIamCredentialsSecret(ibm *providerIBM, secretName *string, secretGroupNa
...
+	if err != nil {
+		return nil, err
+	}
+	if val, ok := secMap[credentialsConst]; ok {
+		var mval []byte
+		mval, err = json.Marshal(val)
+		if err != nil {
+			return nil, err
+		}
+		return mval, nil
```
Comment (Skarlso) [inline pkg/provider/ibm/provider.go:317]:
```suggestion
		mval, err := json.Marshal(val)
		if err != nil {
			return nil, fmt.Errorf("failed to marshal sec map for service credentials secret: %w", err)
		}

		return mval, nil
```

Comment (IdanAdar) [reply pkg/provider/ibm/provider.go:317]: @Shanti-G Do you agree?

**Comment by Shanti-G** [comment]
@Skarlso , here is the evidence of the manual run:
* Secret store:
image


**Case I. External secrets for secret type service_credentials**
* Creating external secret:
image

* Secret successfully synced:
image


**Case II. External secrets for secret type service_credentials with templating**
* Creating external secret:
image
* Secret successfully synced:
image


*** Logs of operator for the externalsecrets created above:**
image

**Review by IdanAdar — APPROVED** [review]

File: pkg/provider/ibm/provider.go (hunk 3 above)
Comment (Shanti-G) [reply pkg/provider/ibm/provider.go:317]: Suggested changes have been addressed in a new PR: https://github.com/external-secrets/external-secrets/pull/2967

**Pull request merged**